packages/babel-parser/test/fixtures/typescript/type-arguments/call-optional-chain/output.json

```diff
@@ -1,9 +1,12 @@
 {
   "type": "File",
-  "start":0,"end":33,"loc":{"start":{"line":1,"column":0},"end":{"line":4,"column":8}},
+  "start":0,"end":49,"loc":{"start":{"line":1,"column":0},"end":{"line":5,"column":15}},
+  "errors": [
+    "SyntaxError: Identifier expected. (5:0)"
```

Comment:
This error message is really confusing, because at `5:0` (which is the position of `s`) _there is_ an identifier.

Instead of `this.raise(TSErrors.IdentifierExpected)`, we should use `this.expect(tt.parenL)` so that it throws `'(' expected`.

Reply:
@nicolo-ribaudo 
Oh I'll change the error starting point and error message to `'(' expected`.

> Instead of this.raise(TSErrors.IdentifierExpected), we should use this.expect(tt.parenL) so that it throws '(' expected.

`this.unexpected(tt.parentL)` would be ignored. There is `this.unexpected` after returning raise, but no error happened and found that's because of the `tsTryParseAndCatch`. So I returned the raise. 🤔 

https://github.com/babel/babel/blob/36d770c844b9bbf2717c24f64be6c690a0d346b3/packages/babel-parser/src/plugins/typescript/index.js#L2093

Reply:
I moved the check part and use `this.unexpected`